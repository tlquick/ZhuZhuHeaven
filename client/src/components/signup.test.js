import { BrowserRouter } from "react-router-dom";
import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Signup from "./signup";
const mockedUsedNavigate = jest.fn();
const AuthService = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));
beforeEach(() => {
  render(
    <BrowserRouter>
      <Signup />
    </BrowserRouter>
  );
});

afterEach(() => {
  cleanup();
});
test("renders signup page", () => {
  const signupElement = screen.getByTestId("signup-1");
  expect(signupElement).toBeInTheDocument();
  expect(signupElement).toContainHTML("img"); //contains a <img> tag
  expect(signupElement).toContainHTML("form"); //contains a <form> tag
});
test("matches signup snapshot", () => {
  const tree = renderer.create(<Signup />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("signin link redirects", () => {
  userEvent.click(screen.getByText(/Sign In/));
  expect(
    screen.getByRole("button", {
      name: /Sign Up/i,
    })
  ).toBeInTheDocument();
});
test("alert when fields are empty", async () => {
  global.alert = jest.fn();
  const buttonElement = screen.getByRole("button");
  await userEvent.click(buttonElement);
  expect(global.alert).toHaveBeenCalledTimes(1);
});
