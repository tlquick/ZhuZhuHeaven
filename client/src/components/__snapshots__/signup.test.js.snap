// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches signup snapshot 1`] = `
<div>
  <div
    className="card card-container"
    data-testid="signup-1"
  >
    <img
      alt="profile-img"
      className="profile-img-card mx-auto"
      src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
    />
    <form
      onSubmit={[Function]}
    >
      <div>
        <div
          className="form-group"
        >
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="form-control"
            name="username"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="sign-button btn btn-primary btn-lg btn-block"
          >
            Sign Up
          </button>
        </div>
      </div>
      <button
        style={
          Object {
            "display": "none",
          }
        }
      />
      <div
        className="link"
      >
        Already registered? 
        <a
          href="/user/signin"
        >
          Sign In
        </a>
      </div>
    </form>
  </div>
</div>
`;
